### PR TITLE
Add code sample to illustrate capability APIs

### LINKED ISSUE #11994: What does "throw from the implementation and offer capability APIs" mean?
In the following section from https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting (with emphasis added on the part I'm not sure on):

>  CONSIDER targeting .NET implementations in addition to .NET Standard.
>
> Targeting .NET implementations allows you to call platform-specific APIs that are outside of .NET Standard.
>
> Do not drop support for .NET Standard when you do this. _Instead, throw from the implementation and offer capability APIs._ This way, your library can be used anywhere and supports runtime light-up of features.

I believe I understand the concept (define a common API, but potentially have different concrete implementations depending on the target), but I was wondering specifically about the wording in "throw from the implementation and offer capability APIs," as that doesn't seem to really jive well with what I thought was being recommended; throw what from what implementation, and what's a capability API and how does it differ from any other type of API?   Am I understanding correctly the gist of what's being recommended, or is there something else here that I'm not getting?

Thanks!

edit by @mairaw: add article details

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5b3638c9-ea3c-1c9e-2763-5c3e8e965663
* Version Independent ID: aff99d6b-9471-9277-160b-670ee8028b53
* Content: [Cross-platform targeting for .NET libraries](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting)
* Content Source: [docs/standard/library-guidance/cross-platform-targeting.md](https://github.com/dotnet/docs/blob/master/docs/standard/library-guidance/cross-platform-targeting.md)
* Product: **dotnet**
* GitHub Login: @JamesNK
* Microsoft Alias: **mairaw**

### PR DESCRIPTION
**Pull request by terrajobst**
Fixes #11994.

### CODE REVIEW

File: docs/standard/library-guidance/cross-platform-targeting.md [hunk 1]
```diff
@@ -53,11 +53,42 @@ To shield your consumers from having to build for individual frameworks, you sho
 >
 > Do not drop support for .NET Standard when you do this. Instead, throw from the implementation and offer capability APIs. This way, your library can be used anywhere and supports runtime light-up of features.
 
+```C#
```
Comment (JamesNK) [inline docs/standard/library-guidance/cross-platform-targeting.md]:
```suggestion
```csharp
```

**Review by JamesNK — APPROVED** [review]

**Comment by mairaw** [comment]
It seems ms.date should also be updated in this case to reflect a meaningful update.

**Comment by terrajobst** [comment]
Will do

________________________________

**Comment by mairaw** [comment]
Took care of that for you 😉 I'll merge when the build finishes.

**Pull request merged**